### DISCUSSION TITLE
Inovelli Aux Switch Scenes Not Interpreted Correctly

### DISCUSSION THREAD

**Discussion by UngluedChalice** [zwave-js/zwave-js#6704] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

**What is happening?**

Multi-tap presses (which trigger scenes) on an aux switch connected to an Inovelli Red 2-1 Dimmer (VZW31-SN) are not interpreted correctly when parameter 123 on the Inovelli is set to `On`. The `value` output is identical to the `value_raw` output. 

Result when parameter 123 is `Off`
```
property_name: scene
  property_key: "001"
  property_key_name: "001"
  value: KeyPressed2x
  value_raw: 3
```
Result when parameter 123 is `On`
```
property_name: scene
  property_key: "004"
  property_key_name: "004"
  value: 3
  value_raw: 3
```

**What did you expect to happen instead?**

The `value` should have the `KeyPressed` appended to the front of it and the `x` on the end. Without this, automations can't interpret it. 
```
property_name: scene
  property_key: "004"
  property_key_name: "004"
  value: KeyPressed2x
  value_raw: 3
```

  1. Go to configuration for the Red 2-1 switch and set parameter 123 to on. 
  2. Listen to `zwave_js_value_notification` and press either the config button (1x, 2x, or ...5x) or press Up or Down on the paddle two to five times to trigger a scene. 
  3. Look at the event data and check to see if the value has the correct `KeyPressed` added on it or not. 

**Anything else we should know?**
This is my first bug report and I'm trying my best to include all the useful info and be clear in the problem, but sometimes I'm not sure what is useful or not. I don't really need this feature, but want to do my part and contribute to make things better. 

### Software versions

zwave-js-ui: 9.9.1
zwave-js: 12.4.4

Home Assistant Z-Wave JS Addon: 3.4.1

-Home Assistant-
Core: 2024.3.3
Supervisor: 2024.03.0
Operating System: 12.1
Frontend: 20240307.0

### Device information

Manufacturer: Inovelli
Model name: VZW31-SN
FW: v1.2.0
SDK: v7.18.1
Node ID: 23, 37 (tried two different switches, both had the same issue. I have a bunch more of these switches on my network as well, though most don't have an aux connected)

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[z-ui_2024-03-25.log](https://github.com/zwave-js/node-zwave-js/files/14751139/z-ui_2024-03-25.log)

[a comment by an automated account is not shown]

**Comment by AlCalzone** [comment]
@UngluedChalice Please re-interview the device once with the parameter ON and once with it OFF and make a **driver log** of the whole process. 
Also check the bot's links again. What you posted is the **application log** which is the section above making a **driver log**.

It could be that the device doesn't report support for scene 004 until the parameter is on, which means the metadata HA uses to lookup the value label is missing.

**Comment by UngluedChalice** [comment]
Ah, the good bot knew I had messed it up and I didn't look carefully enough. Sorry about that!

[zwavejs_2024-04-06.log](https://github.com/zwave-js/node-zwave-js/files/14893885/zwavejs_2024-04-06.log)

This should be the correct log.

**Reply by AlCalzone** [reply]
There's no change in Central Scene capabilities, whether the parameter is on or off. In both cases the switch reports that it supports 3 scenes, not 4:
```
2024-04-06T14:33:09.778Z DRIVER « [Node 038] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -81 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 105
                                    │ security class:  S2_Authenticated
                                    └─[CentralSceneCCSupportedReport]
                                        scene count:                     3
                                        supports slow refresh:           true
                                        supported attributes (scene #1):
                                        · KeyPressed
                                        · KeyReleased
                                        · KeyHeldDown
                                        · KeyPressed2x
                                        · KeyPressed3x
                                        · KeyPressed4x
                                        · KeyPressed5x
                                        supported attributes (scene #2):
                                        · KeyPressed
                                        · KeyReleased
                                        · KeyHeldDown
                                        · KeyPressed2x
                                        · KeyPressed3x
                                        · KeyPressed4x
                                        · KeyPressed5x
                                        supported attributes (scene #3):
                                        · KeyPressed
                                        · KeyReleased
                                        · KeyHeldDown
                                        · KeyPressed2x
                                        · KeyPressed3x
                                        · KeyPressed4x
                                        · KeyPressed5x
```

I guess you'll have to take this to Inovelli's support.

**Comment by InovelliUSA** [comment, marked as answer]
Thanks, we are working on a firmware update to resolve this issue.
